src/Sylius/Bundle/ProductBundle/Form/DataTransformer/VariantToCombinationTransformer.php

```diff
@@ -78,16 +74,14 @@ public function reverseTransform($value)
      */
     private function matches($value)
     {
-        foreach ($this->variable->getVariants() as $variant) {
+        foreach ($this->product->getVariants() as $variant) {
             foreach ($value as $option) {
-                if (null === $option || !$variant->hasOption($option)) {
-                    continue 2;
+                if (null !== $option && $variant->hasOption($option)) {
```

Comment:
There is problem in changed logic. The previous one would match only if all of `values` would be matched with variant options. Now it is enough if any of them is correct.